pkg/scheduler/core/scheduling_queue_test.go

```diff
@@ -473,3 +474,38 @@ func TestUnschedulablePodsMap(t *testing.T) {
 		})
 	}
 }
+
+func TestSchedulingQueue_Close(t *testing.T) {
```

Comment:
It's not clear whether "draining" the queue is supposed to work after it is closed. Does it behave differently for the two implementations?

Reply:
My reading in the source code is both FIFO and PriorityQueue still handle the resting items after it's closed.

Reply:
It is not necessary to process all queue items before closing it. In fact, if provided such a guarantee, we might not be able to shutdown the scheduler for a long time as the rate of addition of pods to the queue could be larger than the rate of scheduling them.

Reply:
Apologize the misleading words. I re-read the code, my previous statement is inaccurate. It should be "Resting items will be handled one time *at most* if queue is closed."

It's because in Pop(), if there are other items enqueued and queue is closed, next item will be popped out. And in outer logic, wait.Until() will return gracefully as stopChan is closed.

Reply:
So I think we're good at dealing with at most one more item.

Reply:
Yes, I believe we are good here, but would like to hear @k82cn 's thoughts too.

Reply:
I'm ok with that :)

The original purpose is to make sure 1. all goroutine are stop, 2. no unexpected action after stop because of stop order; as I suffered such kind of issue in our other prod, a C++ application, e.g. core dump. I'm ok because we did not find any issue right now :)